Use internal SQLite3 database

vovimayhem opened the pull request:
[Closes #50]
[Closes #69]
[Closes #19]

## Background
Replaces the PostgreSQL database with an internal SQLite database, as the data is typically read-only while running on servers, and updates can be deployed as part of the code. Also, it will fix some open issues.

## Features
1. Migrate DB from Postgresql to SQLite.
2. Add an index to ZipCodes to full-text search inside SQLite
3. Fix issues in searches caused by migrating between DBs
4. add items per-page parameter to add flexibility when you require more than 15
5. deployment has set from GCP to Azure, avoiding artifact registry, to save costs, now it uses Dockerhub
6. No more PR deployments to save costs in the cloud
7. upgrade DB from correos de México records

## Conclusion
After those features, we got a renewed API to request the different Zip Codes inside Mexico.

vovimayhem committed:
Removed postgres service

vovimayhem committed:
Replace postgres config with sqlite3

vovimayhem committed:
Fixed failing migrations

vovimayhem commented:
The only thing not working that good is the "zip code" filtering functionality... here are some links with info related to how I'm planning to solve it:
- https://www.sqlite.org/fts5.html
- https://www.sqlitetutorial.net/sqlite-full-text-search
- https://viblo.asia/p/rails-full-text-search-with-sqlite-OeVKBvvEKkW

Following the examples of the first link, I validated that looks like a viable replacement for the "unaccent" extension in PostgreSQL... but will deviate from the "standard" activerecord practice... meaning more SQL...

Wikitelmex commented:
`checking for sqlite3.h... no
sqlite3.h is missing. Try 'brew install sqlite3',
'yum install sqlite-devel' or 'apt-get install libsqlite3-dev'
and check your shared library search path (the
location where your sqlite3 shared library is located).`

That error appeared when I opened the container, Is a good idea to add the `sudo apt install libsqlite3-dev` to the docker file.

wikitelmex1 committed:
fix ILIKE issue in SQLite

wikitelmex1 committed:
generate values

wikitelmex1 committed:
loading data with indexes, next is start searching

wikitelmex1 committed:
state search done, new method alpharize to convert accented characters to alphanumeric

wikitelmex1 committed:
finish fixing zip code searches

wikitelmex1 committed:
go to lunch, commit to continue

wikitelmex1 committed:
fix specs

wikitelmex1 committed:
delete comments and fix format

wikitelmex1 committed:
description about populate db

wikitelmex1 committed:
items per page flexibility added

dvlex committed:
Merge branch 'main' into feature/use-internal-database

dvlex committed:
add to devcontainer sqlite extension

fix instruction in readme file

download the new zip codes database

add to rake task, the loadev task, to load locally the db

dvlex committed:
change the port to 80 to avoid aditional azure config

add dockerignore exception to /tmp/pids folder

dvlex committed:
keep the pids folder, because it causes a failure

dvlex committed:
fix gitignore issue

dvlex committed:
fix deletion of pids folder that causes a issue when build and deploy

dvlex committed:
add ci/cd to dockerhub/azure

dvlex committed:
Update CI & CD workflow to trigger on workflow_dispatch event for test in PR

dvlex committed:
test pr flow

dvlex committed:
change to main

dvlex committed:
fix codeclimate issue try 1

dvlex committed:
try 2

dvlex committed:
try 3

dvlex committed:
try 4

dvlex committed:
try 5

dvlex committed:
try 6

dvlex committed:
try 6

dvlex committed:
try 7

dvlex committed:
try 8
